src/main/resources/static/js/MainWindow.js

```diff
@@ -1,670 +1,769 @@
+		streamData : streams,
+		mapOfMagicNumbers : map,
+		dataType : filetype,
+		frostURL : url
+	};
+
+	var jsoncfg = JSON.stringify(formData, null, 4);
+
+	$.ajax({
+		type : "POST",
+		url : "config/create",
+		contentType : "application/json",
+		data : jsoncfg,
+		dataType : "json",
+		error : function(e) {
+			addToLog(e.responseText);
+		},
+		success : function(e) {
+			var option = new Option(e.name, e.name, null, null);
+			option.setAttribute("data-value", JSON.stringify(e, null, 4));
+			$("#configs").append(option).val(e.name).trigger("change");
+			addToLog("Configuration saved.");
+		}
+	});
 }
 
 /**
  * loads the config with the id id
+ * 
  * @param id
  */
 function loadConfig(id) {
 
-    $.ajax({
-        type: "GET",
-        url: "config/single",
-        data: {
-            configId: id
-        },
-        success: function (response) {
-        	
-        	
-        	
-            mappingData = response.mapOfMagicNumbers;
-            $("#frostserverurl").val(response.frostURL);
-            
-            urlconfirmed(function() {
-            	$("#delimiter").val(response.delimiter);
-                $("#headerlines").val(response.numberOfHeaderlines);
-                currentDelimiter = response.delimiter;
-                currentHeaderLines = response.numberOfHeaderlines;
-                preview();
-
-
-                $("#selecttime").find("option").each(function () {
-                    if ($(this).attr("data-value") === response.timezone) {
-                        $("#selecttime").val($(this).val()).trigger("change");
-                    }
-                });
-
-                var table = $("#timeTable");
-                var lines = response.dateTime.length;
-                var rows = document.getElementById("timeTable").rows.length - 1;
-                while (lines > rows) {
-                    addRow(table, "2", "20");
-                    rows++;
-                }
-                while (lines < rows) {
-                    delLastRow("timeTable");
-                    rows--;
-                }
-
-                var i = 0;
-                table.find("tbody tr").each(function () {
-                    var $td = $(this).find("td");
-                    $td.eq(0).find("input").val(response.dateTime[i].column);
-                    $td.eq(1).find("input").val(response.dateTime[i++].string);
-                });
-
-
-                lines = response.streamData.length;
-                if (lines > 0) {
-                    thingConfig(response.streamData, response.frostURL);
-                }
-            });
-           
-        },
-        error: function (e) {
-            addToLog(e.responseText);
-        }
-    });
+	$
+			.ajax({
+				type : "GET",
+				url : "config/single",
+				data : {
+					configId : id
+				},
+				success : function(response) {
+
+					mappingData = response.mapOfMagicNumbers;
+					$("#frostserverurl").val(response.frostURL);
+
+					urlconfirmed(function() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['urlconfirmed' was used before it was defined. (no-use-before-define)](https://app.codacy.com/gh/FraunhoferIOSB/ChillImport/pullRequest?prid=4781409)